### PR TITLE
ATO-546: Allow cross-account access for identity credentials table

### PR DESCRIPTION
**Pull request by hannah-leong**
## What

Added policies etc to allow for cross-account access to the identity credentials table

## Checklist

- [ ] Impact on orch and auth mutual dependencies has been checked.

### CODE REVIEW

File: ci/terraform/oidc/dynamo-policies.tf [hunk 1]
```diff
@@ -1,5 +1,6 @@
 locals {
-  client_registry_kms_key_arn = data.terraform_remote_state.shared.outputs.client_registry_kms_key_arn
+  client_registry_kms_key_arn      = data.terraform_remote_state.shared.outputs.client_registry_kms_key_arn
+  identity_credentials_kms_key_arn = data.terraform_remote_state.shared.outputs.identity_credentials_kms_key_arn
```
Comment (CarlyG55) [inline ci/terraform/oidc/dynamo-policies.tf:3]: would be nice to pull these into the shared.tf file with other locals (I've moved in my PR)

File: ci/terraform/shared/dynamodb.tf [hunk 2]
```diff
@@ -542,6 +542,12 @@ resource "aws_dynamodb_resource_policy" "client_registry_table_policy" {
   policy       = data.aws_iam_policy_document.cross_account_table_resource_policy_document.json
 }
 
+resource "aws_dynamodb_resource_policy" "identity_credentials_table_policy" {
+  count        = var.identity_credentials_cross_account_access_enabled ? 1 : 0
+  resource_arn = aws_dynamodb_table.identity_credentials_table.arn
+  policy       = data.aws_iam_policy_document.cross_account_table_resource_policy_document.json
```
Comment (CarlyG55) [inline ci/terraform/shared/dynamodb.tf:548]: this policy doc doesn't allow deletion, and one of the policies in the file enough is a deletion policy so a lambda will probably need that

Comment (hannah-leong) [reply ci/terraform/shared/dynamodb.tf:548]: If adding `dynamodb:DeleteItem` to the cross-account policy, then will apply to all tables. As we won't give all lambdas permission to delete, this should be okay but shall have a look into it next week to see if it is possible to just apply it to the identity credentials

Comment (CarlyG55) [reply ci/terraform/shared/dynamodb.tf:548]: Yeah agreed that maybe in this case we want a different policy

**Review by CarlyG55 — APPROVED** [review]

**Review by CarlyG55 — APPROVED** [review]

**Pull request merged**